dm/syncer/validate_worker.go

```diff
@@ -199,12 +264,28 @@ func (vw *validateWorker) validateRowChanges(table *validateTableInfo, rows []*r
 	return res, nil
 }
 
+func (vw *validateWorker) getPendingChangesMap() map[string]*tableChange {
```

Comment:
I recall a recently found data race. This function will return same map to caller, the lock can't protect R/W to the map KV. Maybe the lock is not needed?

Reply:
worker will not write it when we're persisting since there's no rowChange in the worker channel, so there'll be no data race.

lock is a must for concurrent r/w accessed and non-atomic variable, even there'll be only one routine may read it as in this case, to make sure we can see the latest value.

Reply:
OK I'm not sure if golang need a memory barrier 🤔